pkg/service/egress.go

```diff
@@ -236,58 +223,88 @@ func (s *EgressService) StopEgress(ctx context.Context, req *livekit.StopEgressR
 	return info, nil
 }
 
-func (s *EgressService) updateWorker() {
+func (s *EgressService) startWorker() error {
+	rs := s.es.(*RedisStore)
+	if err := rs.Start(); err != nil {
+		logger.Errorw("failed to start redis egress worker", err)
+		return err
+	}
+
 	sub, err := s.rpcClient.GetUpdateChannel(context.Background())
 	if err != nil {
 		logger.Errorw("failed to subscribe to results channel", err)
-		return
+		return err
 	}
 
-	resChan := sub.Channel()
-	for {
-		select {
-		case msg := <-resChan:
-			b := sub.Payload(msg)
-
-			res := &livekit.EgressInfo{}
-			if err = proto.Unmarshal(b, res); err != nil {
-				logger.Errorw("failed to read results", err)
-				continue
-			}
-
-			switch res.Status {
-			case livekit.EgressStatus_EGRESS_ACTIVE,
-				livekit.EgressStatus_EGRESS_ENDING:
-
-				// save updated info to store
-				err = s.es.UpdateEgress(context.Background(), res)
-				if err != nil {
-					logger.Errorw("could not update egress", err)
-				}
-
-			case livekit.EgressStatus_EGRESS_COMPLETE,
-				livekit.EgressStatus_EGRESS_FAILED,
-				livekit.EgressStatus_EGRESS_ABORTED:
-
-				// delete from store
-				err = s.es.DeleteEgress(context.Background(), res)
-				if err != nil {
-					logger.Errorw("could not delete egress from store", err)
+	go func() {
+		resChan := sub.Channel()
+		for {
+			select {
+			case msg := <-resChan:
+				b := sub.Payload(msg)
+
+				res := &livekit.EgressInfo{}
+				if err = proto.Unmarshal(b, res); err != nil {
+					logger.Errorw("failed to read results", err)
+					continue
 				}
 
-				// log results
-				if res.Error != "" {
-					logger.Errorw("egress failed", errors.New(res.Error), "egressID", res.EgressId)
-				} else {
-					logger.Infow("egress ended", "egressID", res.EgressId)
+				ensureRoomName(res)
```

Comment:
is this for backwards compatibility?

Reply:
yeah, left a comment above the new function